# PATCH [ 2 of 2 ] : Replace strtoll_l() with a simple strtoll() call.
# 
# GCC's name-mangling may not work. Change source since Musl doesn't support strtoll_l(), 
# so replace it with a simple strtoll() call. Helps fix "abi tag" errors when compiling libcxx on musl

--- llvm-project-15.0.5.src.orig/libcxx/include/locale	2022-11-16 00:28:29.000000000 -0600
+++ llvm-project-15.0.5.src/libcxx/include/locale	2022-12-23 15:23:41.760570897 -0600
@@ -760,7 +760,8 @@
         typename remove_reference<decltype(errno)>::type __save_errno = errno;
         errno = 0;
         char *__p2;
-        long long __ll = strtoll_l(__a, &__p2, __base, _LIBCPP_GET_C_LOCALE);
+//glibc        long long __ll = strtoll_l(__a, &__p2, __base, _LIBCPP_GET_C_LOCALE);
+	long long __ll = strtoll(__a, &__p2, __base);
         typename remove_reference<decltype(errno)>::type __current_errno = errno;
         if (__current_errno == 0)
             errno = __save_errno;
@@ -800,7 +801,8 @@
         typename remove_reference<decltype(errno)>::type __save_errno = errno;
         errno = 0;
         char *__p2;
-        unsigned long long __ll = strtoull_l(__a, &__p2, __base, _LIBCPP_GET_C_LOCALE);
+//glibc        unsigned long long __ll = strtoull_l(__a, &__p2, __base, _LIBCPP_GET_C_LOCALE);
+	unsigned long long __ll = strtoull(__a, &__p2, __base);
         typename remove_reference<decltype(errno)>::type __current_errno = errno;
         if (__current_errno == 0)
             errno = __save_errno;
